### ISSUE TITLE
/mysql IPC socket connection

### ISSUE THREAD

**Issue by elmarti** [ForbesLindesay/atdatabases#119]
Unfortunately I need to pivot a project to use IPC sockets instead of a URL connection string. It looks like this isn't possible with this library.

### Proposal
Add an alternative connection object parameter which allows easier configuration of connection and the use of IPC sockets. 

See TypeORM's connection options for an example https://github.com/typeorm/typeorm/blob/master/docs/connection-options.md

**Comment by ForbesLindesay** [comment]
Yes, we already support something very similar for Postgres. I'm happy to accept a pull request that allows specifying the connection configuration as an object.
